### PR TITLE
add filter transform

### PR DESCRIPTION
**Pull request by eranra**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -225,6 +225,31 @@ Before the first transform suppose we have the keys `DstAddr` and `SrcAddr`.
...
+
+The filter transform module allows setting rules to remove complete entries from
+the output, or just remove specific keys and values from entries.
+
+For example, suppose we have a flow log with the following syntax:
+```
+{"Bytes":20800,"DstAddr":"10.130.2.2","DstPort":36936,"Packets":400,"Proto":6,"SequenceNum":1919,"SrcAddr":"10.130.2.13","SrcHostIP":"10.0.197.206","SrcPort":3100,"TCPFlags":0,"TimeFlowStart":0,"TimeReceived":1637501832}
+```
+
+The bellow configuration will remove (filter) the entry from the output
```
Comment (jotak) [inline README.md:238]: typo "below"

**Review by jotak — COMMENTED** [review]
lgtm, just a typo :)

File: README.md [hunk 2]
```diff
@@ -225,6 +225,31 @@ Before the first transform suppose we have the keys `DstAddr` and `SrcAddr`.
...
+pipeline:
+  transform:
+    - type: filter
+      filter:
+        rules:
+        - input: SrcPort
+          type: remove_entry_if_exists 
+```
+Using `remove_entry_if_doesnt_exist` in the rule reverses the logic and will not remove the above example entry
+Using `remove_field` in the rule `type` instead, cause in outputting the entry after
```
Comment (KalmanMeth) [inline README.md:250]: s/cause/results/

File: README.md (hunk 1 above)
Comment (eranra) [reply README.md]: fixed

File: README.md (hunk 2 above)
Comment (eranra) [reply README.md]: fixed

File: pkg/pipeline/transform/transform_filter_test.go [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+func TestNewTransformFilterRemoveEntryIfExists(t *testing.T) {
+	newTransform := InitNewTransformFilter(t, testConfigTransformFilterRemoveEntryIfExists)
+	transformFilter := newTransform.(*Filter)
+	require.Len(t, transformFilter.Rules, 1)
+
+	input := test.GetIngestMockEntry(false)
+	output := transformFilter.Transform([]config.GenericMap{input})
+	require.Equal(t, output, []config.GenericMap{})
+}
+
```
Comment (ronensc) [inline pkg/pipeline/transform/transform_filter_test.go:92]: There's a missing test for `RemoveEntryIfDoesntExist`

Comment (eranra) [reply pkg/pipeline/transform/transform_filter_test.go:106]: added

**Review by ronensc — APPROVED** [review]

**Review by KalmanMeth — APPROVED** [review]

**Pull request merged**